### PR TITLE
Vectra AI - update Stream parser and add Hunting Queries

### PR DESCRIPTION
**Pull request by danymello**
Fixes #

 - Fix a bug in the  parser (renaming of 2 attributes)

## Proposed Changes

  - Add Hunting Queries for Vectra AI Metadata

### CODE REVIEW

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - CommandAndControl
+relevantTechniques:
+  - T1071.004
+query: |
+  VectraStream
+  | where metadata_type == "metadata_dns"
+  | extend querylength = strlen(['query'])
+  | summarize count() by querylength, orig_hostname 
+  | sort by querylength desc 
+  | extend HostCustomEntity = orig_hostname, IPCustomEntity = id_orig_h
```
Comment (v-maudan) [inline Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml:22]: @danymello, please check and fix query.
image

File: Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+id: 5e3e6cd2-1fba-44c0-8ddf-ad599a06a9df
+name: Vectra AI - Suspicious Unsecured Credentials Group Policy Preferences"
+description: |
+  'Query searches for suspicious Unsecured Credental using GPP Attack.
```
Comment (v-maudan) [inline Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml:4]: Typo Credental -> Credential

**Review by v-maudan — CHANGES_REQUESTED** [review]

**Comment by v-rucdu** [comment]
@ashwin-patil : Could you please review the queries?

Thanks!!!

**Comment by shainw** [comment]
@v-rucdu - Adding Ajeet to this one also.

File: Solutions/Vectra/Hunting Queries/VectraMaliciousToolFileCopy.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+requiredDataConnectors:
+  - connectorId: AIVectraStream
+    dataTypes:
+      - VectraStream
+tactics:
+  - CredentialAccess
+  - LateralMovement
+  - Collection
+relevantTechniques:
+  - T1557.001
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraMaliciousToolFileCopy.yaml:16]: Wondering if it also makes sense to have Technique T1021.002 here?

File: Solutions/Vectra/Hunting Queries/VectraPotentialDCSync.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+id: 489d1c70-5fe2-426b-afbe-371ab58605fe
+name: Vectra AI - Potential DCSync Attack"
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraPotentialDCSync.yaml:2]: Please remove the " in the end.

File: Solutions/Vectra/Hunting Queries/VectraSuspectedKali.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+id: 98ce0ffc-eb1d-458e-835f-6c72ff4d53d1
+name: Vectra AI - Possible Kali Linux Detected"
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraSuspectedKali.yaml:2]: Please remove the " in the end.

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousBeacon.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+id: b8661942-b3ba-48a4-ae9d-db7c326ebd46
+name: Vectra AI - Beaconing Behaviors"
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraSuspiciousBeacon.yaml:2]: Please remove the " in the end.

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousNbOfSubDomains.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+id: b8bacbda-3faa-49fb-b000-9334b2b51c8e
+name: Vectra AI -
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraSuspiciousNbOfSubDomains.yaml:2]: Wondering if you missed giving a complete name to this.

File: Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+id: 5e3e6cd2-1fba-44c0-8ddf-ad599a06a9df
+name: Vectra AI - Suspicious Unsecured Credentials Group Policy Preferences"
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml:2]: Please remove the " in the end.

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+description: |
+  'Query searches long DNS queries.
+  A high volume of long DNS queries may indicate the usage of DNS Tunneling.
+  Metadata required = metadata_dns'
+severity: High
+requiredDataConnectors:
+  - connectorId: AIVectraStream
+    dataTypes:
+      - VectraStream
+tactics:
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml:12]: Should we also have Exfiltration in here.

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  - CommandAndControl
+relevantTechniques:
+  - T1071.004
```
Comment (aprakash13) [inline Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml:15]: Should we also have T1048.003 in here.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Made a few suggestions for changes. Can you please have a look. Thanks

File: Solutions/Vectra/Hunting Queries/VectraHighVolumeOfNXDOMAIN.yaml [hunk 11]
```diff
@@ -0,0 +1,32 @@
+id: 9e8038a8-926f-48cd-9075-5f69b15f5192
+name: Vectra AI - Potential Exfiltration over DNS"
```
[the remark this answers is not in the document]
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraHighVolumeOfNXDOMAIN.yaml:2]: implemented

File: Solutions/Vectra/Hunting Queries/VectraHighVolumeOfNXDOMAIN.yaml [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  Metadata required = metadata_dns'
+severity: High
+requiredDataConnectors:
+  - connectorId: AIVectraStream
+    dataTypes:
+      - VectraStream
+tactics:
+  - Exfiltration
+relevantTechniques:
+  - T1048.003
```
[the remark this answers is not in the document]
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraHighVolumeOfNXDOMAIN.yaml:14]: yes, I can add that. The query is targeted to detect Exfiltration but DGA can potentially lead also lead to many NXDomain DNS responses.

File: Solutions/Vectra/Hunting Queries/VectraMaliciousToolFileCopy.yaml (hunk 3 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraMaliciousToolFileCopy.yaml:16]: Indeed, not pointing to the right tactic. it should map to T1570

File: Solutions/Vectra/Hunting Queries/VectraPotentialDCSync.yaml (hunk 4 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraPotentialDCSync.yaml:2]: implemeted

File: Solutions/Vectra/Hunting Queries/VectraSuspectedKali.yaml (hunk 5 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraSuspectedKali.yaml:2]: implemeted

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousBeacon.yaml (hunk 6 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraSuspiciousBeacon.yaml:2]: implemeted

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousNbOfSubDomains.yaml (hunk 7 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraSuspiciousNbOfSubDomains.yaml:2]: yes. I completed it and adjust the query to have a threshold.

File: Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml (hunk 8 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraUnsecuredCredsGPP.yaml:2]: implemented

File: Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml (hunk 10 above)
Comment (danymello) [reply Solutions/Vectra/Hunting Queries/VectraSuspiciousLongDNSQueries.yaml:15]: yes. added

**Comment by danymello** [comment]
> Made a few suggestions for changes. Can you please have a look.
> 
> Also for all of these hunting queries when we are mapping entities in the end, I was wondering if we have a TimeGenerated field that we can map as well as an entity.
> https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#old-entity-mapping-method
> 
> Thanks.

I added 'timestamp = TimeGenerated' for query which does not use aggregation.

**Review by aprakash13 — APPROVED** [review]
The suggestions have been implemented by @danymello.

**Pull request merged**